src/components/cl/hier/cl_hier_context.c

```diff
@@ -23,6 +24,13 @@ UCC_CLASS_INIT_FUNC(ucc_cl_hier_context_t,
 
     UCC_CLASS_CALL_SUPER_INIT(ucc_cl_context_t, cl_config,
                               params->context);
+    if (params->params.oob.n_oob_eps == 1) {
```

Comment:
@vspetrov - based on our discussion in UCC WG call, this check might not be required. However, I got the error below during context creation (some pytorch applications).
```
ucc_context.c:676  UCC  ERROR failed to init ctx topo
```

I think we still need this check here to skip creating CL/HIER ctx at all, please let me know if you think otherwise Since OOB is not always available, would it be better to check `params->context->addr_storage.size` instead?

Reply:
cc @vspetrov @Sergei-Lebedev - follow up the question above

Reply:
resolved as no need to check here